pkg/reconciler/knativeserving/common/validation.go

```diff
@@ -0,0 +1,276 @@
+		for _, con := range deploy.Spec.Template.Spec.Containers {
+			if con.Name == deploy.Name {
+				img := strings.Split(con.Image, ":")
+				if len(img) != 2 {
+					log.V(1).Info("version check", "failed to get version from "+con.Image)
+					return false, fmt.Errorf("failed to get version information")
+				}
+				ver = img[1]
+			}
+		}
+	}
+
+	log.V(1).Info(depName+" version check", "installed version->", ver, "version expected",
+		strings.Join(verlist, ","))
+	return checkDependencyVersion(&ver, verlist), nil
+}
+
+// PodExec takes command and run the command in the specified container of Pod
+func podExec(client kubernetes.Interface, config *rest.Config, podNamespace, podLabel,
+	container string, command []string) (*bytes.Buffer, *bytes.Buffer, error) {
+	pods, err := client.CoreV1().Pods(podNamespace).List(metav1.ListOptions{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		LabelSelector: podLabel,
+	})
+	if err != nil || len(pods.Items) < 1 {
+		log.V(1).Info("Dependency Validation", "failed to get pods with label", podLabel)
+		return nil, nil, err
+	}
+
+	log.V(1).Info("Dependency Validation", "pod", pods.Items[0].Name)
+	req := client.CoreV1().RESTClient().Post().Resource("pods").Name(pods.Items[0].Name).Namespace(podNamespace).
+		SubResource("exec").
+		VersionedParams(&v1.PodExecOptions{
+			Command:   command,
+			Container: container,
+			Stdin:     false,
+			Stdout:    true,
+			Stderr:    true,
+			TTY:       false,
+		}, k8sscheme.ParameterCodec)
+
+	exec, err := remotecommand.NewSPDYExecutor(config, "POST", req.URL())
+	if err != nil {
+		return nil, nil, err
+	}
+	var stdout, stderr bytes.Buffer
+	err = exec.Stream(remotecommand.StreamOptions{
+		Stdin:  nil,
+		Stdout: &stdout,
+		Stderr: &stderr,
+		Tty:    false,
+	})
+
+	return &stdout, &stderr, err
+}
+
+func getIstioVersion(client kubernetes.Interface, config *rest.Config) (string, error) {
+	cmds := []string{"/bin/bash", "-c",
+		"env | grep ISTIO_META_ISTIO_VERSION | awk -F'=|\\n' '{print $2}'"}
+	stdout, stderr, err := podExec(client, config, istioNamespace, istioPilotPodLabel,
+		istioPilotContainer, cmds)
+
+	if err != nil {
+		if stderr.String() != "" {
+			return "", fmt.Errorf("error execing get istio version: %v\n%s", stderr.String(), err)
+		}
+		return "", err
+	}
+
+	log.V(1).Info("Dependency Validation", "istio version from podExec", stdout.String())
+	return stdout.String(), nil
+}
+
+func validate(client kubernetes.Interface, config *rest.Config, t *validateTask) error {
+	basemsg := t.meta.name + " validation"
+	// check namespace created
+	_, err := client.CoreV1().Namespaces().Get(t.meta.namespace, metav1.GetOptions{})
+	if err != nil {
+		msg := t.meta.namespace + " doesn't exist"
+		log.Error(err, basemsg, "namespace checking", msg)
+		return fmt.Errorf(msg)
+	}
+
+	// for istio, version could be get frome env of container istio-proxy in pod istio-pilot
```

Comment:
<!--typo-bot-->
```suggestion
	// for istio, version could be get from env of container istio-proxy in pod istio-pilot
```

Reply:
> I would think easy access to dependency version mappings is a question for knative-serving itself to answer, no? The operator would consume that information just like a human would.

@jcrossley3 I didn't find information about how to read kn serving mapping with dependency version by operator. Do you have any advice? Or did I misunderstand your comments?